editor/components/block-settings-menu/shared-block-settings.js

```diff
@@ -55,6 +71,7 @@ export default compose( [
 		const { getBlock, getSharedBlock } = select( 'core/editor' );
 		const block = getBlock( uid );
 		return {
+			block,
```

Comment:
Passing the block object will cause more rerenders than necessary (because the object reference changes semi-frequently). We don't need the entire object, we just need its name. There's even a selector for this: `getBlockName`. Similarly, the component doesn't need `sharedBlock` (an object whose reference changes), it just needs to know whether the block is shared or not (a boolean unlikely to change very often). In both cases, we're also being more truthful to the requirements of the component.